coderd/userauth_test.go

```diff
@@ -1532,6 +1532,43 @@ func TestUserLogout(t *testing.T) {
 	}
 }
 
+// TestOIDCSkipIssuer verifies coderd can run without checking the issuer url
+// in the OIDC exchange. This means the CODER_OIDC_ISSUER_URL does not need
+// to match the id_token `iss` field, or the value returned in the well-known
+// config.
+func TestOIDCSkipIssuer(t *testing.T) {
```

Comment:
I'd like to see an explicit test for an issuer mismatch if we're adding this functionality.

Reply:
That is what this does! I'll add a comment to be more clear, but the url hit is `localhost:<port>`, and the issuer in the `well-known` and the jwt are `https://primary.com`

---

Just realized what I think you meant. I'll add a 3 way mismatch to the test, since the values really don't matter with this enabled.

Reply:
I mean, if the configuration option is not enabled, then an issuer mismatch should result in a failure to login (current behaviour).

Reply:
Ahh, will do :+1:

Reply:
https://github.com/coder/coder/blob/c28b912c5b4c0f565df7859decec43f5e2a62efb/coderd/userauth_test.go#L1229-L1243